src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -909,14 +1304,29 @@ static bool processAreaScope(const Scope* scoP, BSONObj &areaQuery)
 * FIXME P9: this function currently abuses of the std::string() wrapping for char* in order to
 * make comparisons work. An smarter solution could be developed.
 *
+* FIXME P7: actually, this functions behaviour depends on the presence of the cerP parameter, in particular
+*
+* - if cerP is NULL, then the output of this function is a set of BSON filters (in the filters vector) to
+*   be applied on MongoDB. The return value is not used. This is the typical use for queryContext and derived
+*   operations
+* - if cerP is not NULL, then the filters are evaluated in memory on the cerP. The return value is used to
+*   specify if cerP match the filter or not. The filters parameter is not used (although the caller needs to
+*   provide it in order to conform with function signature).
+*
+* I don't like too much to have a function with two quite parallel behaviours in the same logic. Probably a
+* smarter design will be to divide this function in two part: the one focused in string parsing and the one
+* focused in appliying filters to each different sytem (either in memory or as BSON).
+*
```

Comment:
Opinions on the above comment and ideas to improve the current implementation are welcome :)